client/src/Visit.tsx

```diff
@@ -32,17 +35,45 @@ export const Visit = (): JSX.Element => {
     return meetingLinkModel;
   };
 
+  const _getRoomLocator = (meetingLink: string): RoomCallLocator | undefined => {
+    let meetingLinkModel: RoomCallLocator | undefined = undefined;
+    // try extracting rooms meeting link from the url
+    try {
+      meetingLinkModel = getRoomsMeetingLink(meetingLink);
+    } catch (error) {
+      meetingLinkModel = undefined;
+    }
+    return meetingLinkModel;
+  };
+
+  const _getParticipantId = (meetingLink: string): string | undefined => {
+    let participantId: string | undefined = undefined;
+    //try extracting rooms participant Id from the url
+    try {
+      participantId = getRoomsUserId(meetingLink);
+    } catch (error) {
+      participantId = undefined;
+    }
+    return participantId;
+  };
+
   // handle going to previous/next page of window history
   window.onpopstate = () => {
     window.location.assign(window.location.href);
   };
 
   const [config, setConfig] = useState<AppConfigModel | undefined>(undefined);
   const [token, setToken] = useState<CommunicationUserToken | undefined>(undefined);
+  const [roomsToken, setRoomsToken] = useState<string | undefined>(undefined);
+  const [userRole, setUserRole] = useState<RoomParticipantRole | undefined>(undefined);
```

Comment:
Instead of adding many different new states, can we combine the Rooms states into one object? I think it would be a better way of organizing these properties since they are all related.
- RoomId, UserId, UserRole, etc.